### PR TITLE
billing for one metrics

### PR DESCRIPTION
**Pull request by vijaykanthm28**

### CODE REVIEW

File: events/alerts/mailgun.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ const (
 	SNAPSHOTTED
 	RUNNING
 	FAILURE
+	LOWFUND
```
Comment (indykish) [inline events/alerts/mailgun.go]: @vijaykanthm28  Make this INSUFFICIENT_FUND

File: events/alerts/mailgun.go [hunk 2]
```diff
@@ -56,6 +57,8 @@ func (v *EventAction) String() string {
 		return "invite"
 	case BALANCE:
 		return "balance"
+	case LOWFUND:
+		return "lowfund"
```
Comment (indykish) [inline events/alerts/mailgun.go]: @vijaykanthm28  Make this INSUFFICIENT_FUND

File: events/alerts/mailgun.go [hunk 3]
```diff
@@ -161,6 +163,8 @@ func subject(eva EventAction) string {
 		sub = "Lets party!"
 	case BALANCE:
 		sub = "Piggy bank!"
+	case LOWFUND:
+		sub = "Piggy bank!"
```
Comment (indykish) [inline events/alerts/mailgun.go]: sub = "Insufficient funds"

File: events/bill.go [hunk 4]
```diff
@@ -37,6 +37,11 @@ func (self *Bill) Watch(eventsChannel *EventChannel) error {
 					if err != nil {
 						log.Warningf("Failed to process watch event: %v", err)
 					}
+				case event.EventAction == alerts.LOWFUND:
+					err := self.alertMail(event)
```
Comment (indykish) [inline events/bill.go]: change method name to `insufficientFund`

File: events/bill.go [hunk 5]
```diff
@@ -61,6 +66,16 @@ func (self *Bill) skip(k string) bool {
 	return !strings.Contains(self.piggyBanks, k)
 }
 
+func (self *Bill) alertMail(evt *Event) error {
```
Comment (indykish) [inline events/bill.go]: change method name to `insufficientFund`

File: events/bills/balances.go [hunk 6]
```diff
@@ -56,6 +57,9 @@ func (b *Balances) String() string {
 //We need add a Notifier interface duck typed by Box and Carton ?
 func NewBalances(id string, m map[string]string) (*Balances, error) {
 	b := &Balances{}
+	if id == "" {
```
Comment (indykish) [inline events/bills/balances.go]: why do we need this check of `id` ?

File: utils/constants.go [hunk 7]
```diff
@@ -196,6 +197,8 @@ const (
 	// it should transition shortly to a more specific status
 	StatusLaunching = Status(LAUNCHING)
 
+	StatusBalanceVerified = Status(BALANCECHECK)
```
Comment (indykish) [inline utils/constants.go]: What do yo do with this flag `BALANCECHECK`

**Comment by indykish** [comment]
@vijaykanthm28  Please close the comments before we merge.

File: utils/constants.go (hunk 7 above)
Comment (vijaykanthm28) [inline utils/constants.go]: @indykish  
to update status after verify balance check

**Comment by indykish** [comment]
@vijaykanthm28  Good job in hacking it so fast. !

**Pull request merged**